components/raftstore/src/store/peer.rs

```diff
@@ -2282,6 +2292,10 @@ where
             }
             // Pause `read_progress` to prevent serving stale read while applying snapshot
             self.read_progress.pause();
+
+            if self.unsafe_recovery_state.is_some() {
+                self.unsafe_recovery_maybe_finish_wait_apply(ctx, /*force=*/ false);
```

Comment:
I think this should be called after `advance_apply_to` in `check_snap_status`

Reply:
here doesn't indicate the snapshot is applied. Seems the test isn't failed, please update the test to cover the case. Maybe use `cluster.get_engine()` to check the data.

Reply:
Updated. 
As for the test, I don't think we need to do any extra work, since, after the "check" logic is changed, `assert_eq!(pd_client.must_get_store_reported(&nodes[1]), 0);` already covers the case.